File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/cert/impl/SwlMemCertManager.java
package i2f.swl.cert.impl;

import i2f.swl.cert.SwlCertManager;
import i2f.swl.cert.data.SwlCert;

import java.util.concurrent.ConcurrentHashMap;

/**
 * @author Ice2Faith
 * @date 2024/8/8 21:10
 * @desc
 */
public class SwlMemCertManager implements SwlCertManager {
    public static final String SERVER_SUFFIX = ".server";
    public static final String CLIENT_SUFFIX = ".client";

    private ConcurrentHashMap<String, SwlCert> certMap = new ConcurrentHashMap<>(1024);

    public String getCertName(String certId, boolean server) {
        if (server) {
            return certId + SERVER_SUFFIX;
        }
        return certId + CLIENT_SUFFIX;
    }

    @Override
    public SwlCert load(String certId, boolean server) {
        String certName = getCertName(certId, server);
        SwlCert ret = certMap.get(certName);
        if (ret != null) {
            return ret.copy();
        }
        return null;
    }

    @Override
    public void store(SwlCert cert, boolean server) {
        if (cert == null) {
            return;
        }
        String certId = cert.getCertId();
        String certName = getCertName(certId, server);
        certMap.put(certName, cert.copy());
    }
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/cert/impl/SwlMemCertManager.js
import SwlCertManager from "../SwlCertManager";

/**
 * @return {SwlMemCertManager}
 * @extends {SwlCertManager}
 */
function SwlMemCertManager() {
    /**
     * 继承父类属性
     */
    SwlCertManager.call(this)
    /**
     *
     * @type {Map|Object}
     */
    this.certMap=new Map()
}

// 继承
SwlMemCertManager.prototype = Object.create(SwlCertManager.prototype)
SwlMemCertManager.prototype.constructor = SwlMemCertManager

/**
 * @return {string}
 */
SwlMemCertManager.SERVER_SUFFIX=function(){
    return ".server"
}

/**
 * @return {string}
 */
SwlMemCertManager.CLIENT_SUFFIX=function(){
    return ".client"
}

/**
 * @param certId {String}
 * @param server {boolean}
 * @return {String}
 */
SwlMemCertManager.prototype.getCertName = function (certId,server) {
    if (server) {
        return certId + SwlMemCertManager.SERVER_SUFFIX();
    }
    return certId + SwlMemCertManager.CLIENT_SUFFIX();
}


/**
 * @param certId {String}
 * @param server {boolean}
 * @return {SwlCert}
 */
SwlMemCertManager.prototype.load = function (certId,server) {
    let certName = this.getCertName(certId, server);
    let ret = this.certMap[certName];
    if (ret != null) {
        return ret.copy();
    }
    return null;
}
/**
 * @param cert {SwlCert}
 * @param server {boolean}
 * @return {void}
 */
SwlMemCertManager.prototype.store = function (cert,server) {
    if (cert == null) {
        return;
    }
    let certId = cert.certId;
    let certName = this.getCertName(certId, server);
    this.certMap[certName]=cert.copy();
}

export default SwlMemCertManager
